keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -233,6 +234,16 @@ interface ArtifactRepository : PeriodicallyCheckedRepository<DeliveryArtifact> {
    */
   fun deletePin(deliveryConfig: DeliveryConfig, targetEnvironment: String, reference: String)
 
+  /**
+   * Given artifact details and promotion status, return its last deployed version
+   */
+  fun getGitMetadataByPromotionStatus(
+    deliveryConfig: DeliveryConfig,
+    environmentName: String,
+    artifactReference: String,
```

Comment:
Consider passing the `DeliveryArtifact` object here instead. I'm finding so many places in my PR where we need to do unnecessary joins between tables just because the methods take only an identifier and not the whole object. This would avoid adding to that problem.

Reply:
can you elaborate on why using `DeliveryArtifact` here will be better? for this function, I need the `deliveryConfig` abject as a whole, which I don't have in `DeliveryArtifact`.

Reply:
I think luis is suggesting instead of artifact reference passing in the whole artifact object. However, since we need the artifact UUID, I don't think it matters here.

Reply:
Going to keep it as is for now

Reply:
Yes, that is what I was suggesting.